rbx_xml/src/deserializer/error.rs

```diff
@@ -0,0 +1,77 @@
+use rbx_dom_weak::types::VariantType;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+#[error(transparent)]
+pub struct DecodeError(Box<ErrorKind>);
```

Comment:
Does this need to be boxed?

Reply:
Technically no but `ErrorKind` is a very big struct and errors are being passed around by every function. It saves us a bunch of space on the stack to have it boxed.